src/d3a/d3a_core/simulation.py

```diff
@@ -289,12 +296,11 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
                     sleep(abs(tick_lengths_s - realtime_tick_length))
 
             self._update_and_send_results()
-            if self.export_on_finish:
+            if self.export_on_finish or self.redis_connection.is_enabled():
```

Comment:
Can you please share the reason for this change? The final version of the files is already send to d3a-web by `_update_and_send_results`, right?

Reply:
The reason for that is that the csv files need to be generated in addition to when the export_on_finish is set to true, also whenever there is a Redis connection and we want to send the zipped version of the files. In the old state of the if clause, the json files were not generated at all, because the parameter export_on_finish was set to false.

Reply:
Ah yes! You are right. Now I understand. Thanks!